storage/ctab/src/main/java/org/openscience/cdk/io/MDLV3000Reader.java

```diff
@@ -405,9 +478,15 @@ public void readAtomBlock(IAtomContainer readData) throws CDKException {
                 }
                 // parse the element
                 String element = tokenizer.nextToken();
-                if (isotopeFactory.isElement(element)) {
-                    atom.setSymbol(element);
-                    isotopeFactory.configure(atom); // ?
+                Elements e = Elements.ofString(element);
+                if (e != Elements.Unknown) {
+                    atom.setAtomicNumber(e.number());
+                } else if ("D".equals(element) && optHydIso.isSet()) {
```

Comment:
I was thinking that the String comparison must be slower than the bool matching, but then I realize that the former is pretty rare, and overall this order must be more efficient...